service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -468,7 +470,7 @@ private JInvocation createRequest(ServiceIDL serviceIdl, JCodeModel jCodeModel,
                 builderInvocation = builderInvocation.invoke("setRequestObject").arg(JExpr.ref(requestObjectName));
             }
 
-            if (endPoint.isIncludeMetaInfo()) {
+            if (endPoint.isIncludeMetaInfo() && endPoint.getMetaInfoMeta() != null && endPoint.getMetaInfoMeta().isDynamicCommandName()) {
```

Comment:
divide this into two conditions, one for endPoint.isIncludeMetaInfo() and the other for the rest. Will make it easier to add other features

Reply:
this code block is just for generating setCommandName method. So I don't see any difference in dividing into 2 conditions or not. If we have other methods which uses this meta in future, let them have this check separately.

Reply:
so that the same condition is not evaluated again and again. also so that the code dealing with metaInfo is in the same place